contracts/test/ERC20.t.sol

```diff
@@ -124,6 +139,30 @@ contract ERC20BaseTest is TestUtils {
         }
     }
 
+    function testFuzz_transferFrom_infiniteApproval(address recipient_, uint256 amount_) public {
+        amount_                  = constrictToRange(amount_,   0, type(uint256).max);
+        uint256 infiniteApproval = type(uint256).max;
+        if (amount_ > infiniteApproval) return;  // Owner must approve for more than amount.
+
+        ERC20User owner = new ERC20User();
+
+        _token.mint(address(owner), amount_);
+        owner.erc20_approve(address(_token), self, infiniteApproval);
+
+        assertTrue(_token.transferFrom(address(owner), recipient_, amount_));
+
+        assertEq(_token.totalSupply(), amount_);
+
+        assertEq(_token.allowance(address(owner), self), infiniteApproval);
+
+        if (address(owner) == recipient_) {
+            assertEq(_token.balanceOf(address(owner)), amount_);
```

Comment:
You can move this check out of the if condition since it's identical in both cases.